.github/workflows/my-ci-pipeline.yml

```diff
@@ -16,3 +16,39 @@ jobs:
         - run: docker build --target test --tag todo_app:test .
         - run: docker run --env-file=.env.template todo_app:test todo_app/tests/unit
         - run: docker run --env-file=.env.template todo_app:test todo_app/tests/integration
+    push_to_registry:
+      needs: build
+      name: Push Docker image to Docker Hub
+      runs-on: ubuntu-latest
+      steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: rogvan/todoapp:prod
+      
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: rogvan/todoapp:prod
```

Comment:
I think you're missing a `target` command here - your app is currently down because Azure is actually just running your tests every time it tries to start! Without the target command, `docker build ...` defaults to building the final stage, which for you is your tests

Reply:
When you build the image locally you set a target e.g. in the below command you use `development` to specify which stage of your Dockerfile to use
```
docker build --target development --tag todo-app:dev .
```

Without a target, the docker build will default to the last defined stage, which for us is the tests, which is why your app isn't currently loading in Azure and we see output like the below in the logs:

>2023-06-26T13:01:16.619160867Z ============================= test session starts ==============================
2023-06-26T13:01:16.619197368Z platform linux -- Python 3.10.10, pytest-7.2.2, pluggy-1.0.0
2023-06-26T13:01:16.619202868Z rootdir: /app
2023-06-26T13:01:16.619206768Z collected 5 items
2023-06-26T13:01:16.619210568Z
2023-06-26T13:01:16.980349343Z todo_app/tests/integration/integration_test.py .                         [ 20%]
2023-06-26T13:01:17.100869971Z todo_app/tests/unit/viewModel_test.py ....                               [100%]
2023-06-26T13:01:17.100884171Z
2023-06-26T13:01:17.100888871Z ============================== 5 passed in 5.33s ===============================

As an aside, it looks like your pipeline is currently erroring because Poetry has moved their install script location - the latest doc command suggests:
```
curl -sSL https://install.python-poetry.org | python3 -
```